### ISSUE TITLE
Explain how to test it on my repository

### ISSUE THREAD

**Issue by spl** [FantasticFiasco/action-update-license-year#39]
**Is your feature request related to a problem? Please describe.**

This project is a great idea! I'm hoping it will save me (and others) time/worry. But first, I'd like to test it out by triggering it on my repository.

**Describe the solution you'd like**

Can you explain how to, say, trigger `action-update-license-year` to determine if it worked correctly? If I want to use it, I'd like to know that it worked before January of next year.

Here's the kind of scenario I'm thinking about:

1. I create a PR to my repo with an action that, when run immediately, will create a PR to update the copyright.
2. Then, depending on whether `action-update-license-year` worked, I'm faced with a decision:
   * Satisfied, I go to 3.
   * Dissatisfied, I file a bug report with you and point you to my failed PR. Hopefully, the error is enough to help you fix any problem. Once fixed, I re-run `action-update-license-year` and return to 2.
3. I change the action in my PR to schedule it as you've shown in your README.

Ideally, you could update your README with how to work through a scenario such as the above. This would help potential users other than myself. Thanks!

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

**Comment by FantasticFiasco** [comment]
I’ll see if I cannot create a repository for you, and I’ll see what I can do about the lacking documentation. I’ll get back to you when I get home from work.

**Comment by FantasticFiasco** [comment]
Clearly there's not enough information in `README.md`, and I'll have to do a better job there, but until then this is what you can do to test the action.

Start with making sure that your current licence file is inaccurate, i.e. isn't specifying 2020. Then create a new workflow in your repository. The name of the YAML file is unimportant but you have to place it in the directory `.github/workflows/`. Add the following content to the file.

```yml
name: Update copyright year(s) in license file

on: push

jobs:
  run:
    runs-on: ubuntu-latest
    steps:
      - name: Update license copyright year(s)
        uses: FantasticFiasco/action-update-license-year@v1
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
```

A commit to your repository will trigger the action, and about a half of a minute after you pushed your code there should be a PR waiting for you. I just tested this in a new clean repo, you can find it [here](https://github.com/FantasticFiasco/action-update-license-year-push-trigger).

**Comment by spl** [comment]
Fantastic! Thanks for the explanation.

I created a repository with only a README, added the action, and ran into an [error](https://github.com/spl/action-update-license-year/actions/runs/152728093):

```
Update license copyright year(s)1s
  Run FantasticFiasco/action-update-license-year@v1
    with:
      token: ***
##[error]Action failed with error: HttpError: Not Found
```

Let me know if I made a mistake somewhere.

**Comment by FantasticFiasco** [comment]
That might be an error on my part. I've only tested the action on repos with a `LICENSE` file in the root. Do you have a license file in your repo?

**Comment by spl** [comment]
I just added a `license.md` to the same repository and encountered another [error](https://github.com/spl/action-update-license-year/runs/822595118):

```
##[error]Action failed with error: HttpError: Reference already exists
```

**Comment by spl** [comment]
Oh, right. I've recently started preferring `license.md` over `LICENSE`. The file content is nearly the same, however.

**Comment by FantasticFiasco** [comment]
> ... and ran into an error:

I've created #42 and will probably have time to look at it tonight.

**Comment by FantasticFiasco** [comment]
> Oh, right. I've recently started preferring license.md over LICENSE. The file content is nearly the same, however.

Was the error due to the fact that the license file with name `LICENSE` still didn't exist, or because the PR was open. I'm not sure I've tested to run the action with an existing PR open.

Clearly I need to write more tests. I have good coverage at the code updating the license file, but am clearly lacking in the interaction with GitHub.

**Comment by FantasticFiasco** [comment]
Would you benefit from being able to configure the name of the license file? If so I'll create a new issue describing the feature request. You can also create it if you would like updates on the issue.

**Comment by spl** [comment]
> Would you benefit from being able to configure the name of the license file? If so I'll create a new issue describing the feature request. You can also create it if you would like updates on the issue.

Yes, I think that would be a useful feature. I'll create it, since I have another idea about it.

**Comment by FantasticFiasco** [comment]
I'm working towards fixing the reported bugs, but won't get there today. I'll continue to work on in the upcoming week. I'll get back to you when it's been done. Thank you for reporting the issue!

**Comment by spl** [comment]
No problem! The next year is exactly 6 months away. 😉

**Comment by FantasticFiasco** [comment]
A new version has now been released, and let me start by apologizing for the time it took. I got sidetracked by simplifying my toolchain, and did a lot to improve the test. I think all bugs we've seen have been fixed.

When it comes to improving the documentation, i.e. this issue, that's my next task. Regarding improving the API and providing options for non-default repositories, e.g. mono-repos or as in your case where the license file is named differently, I think I will start with that after my summer vacation. I have a couple ideas, and will update the relevant issue when the time comes.

Until then, have a really nice summer and keep safe.
